gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/service/util/MainClassResolver.java

```diff
@@ -36,7 +38,22 @@ public static String resolveMainClass(Project project) {
                 .getByType(org.gradle.api.plugins.JavaPluginExtension.class)
                 .getSourceSets()
                 .getByName("main");
-        Set<Path> javaFilesWithMainMethods = main.getAllSource().getSrcDirs().stream()
+
+        Set<File> sourceDirs = new HashSet<>(main.getAllSource().getSrcDirs());
+
+        File generatedDir = project.getBuildDir()
+                .toPath()
+                .resolve("generated")
+                .resolve("sources")
+                .toFile();
+        if (generatedDir.exists() && generatedDir.isDirectory()) {
+            File[] subDirs = generatedDir.listFiles(File::isDirectory);
+            if (subDirs != null) {
+                sourceDirs.addAll(Arrays.asList(subDirs));
+            }
+        }
```

Comment:
See internal conversation about just using `distribution { mainClass 'className' }` to do this manually.

This code doesn't necessarily handle all generated code and also does not correctly pass on task dependencies from code generation tasks to consumers of this `mainClass` data (and this can be tricky to get right) - it would need some more work. I would just use the manual version for now and if other projects need to find the main class in generated we can look at getting this to work properly.